packages/pigeon/lib/objc_generator.dart

```diff
@@ -568,7 +568,8 @@ String _dictGetter(
     if (prefix != null) {
       className = '$prefix$className';
     }
-    return '[$className fromMap:GetNullableObject($dict, @"${field.name}")]';
+    final String mapExpression = 'GetNullableObject($dict, @"${field.name}")';
+    return '($mapExpression ? [$className fromMap:$mapExpression] : nil)';
```

Comment:
Duplicating evaluation of the whole expression in the common case seems needlessly inefficient. Why not just make `fromMap:nil` return `nil`, which is less generated code and more efficient at runtime than this?

Reply:
The new method should be used to simplify line 925 as well.

Reply:
objc_generator.dart:925?  That's the definition of the new method.  You talking about somewhere else?

Reply:
Sorry, I was on a phone and confused myself about the context while searching for the other location.

I'm still not clear on why we can't replace https://github.com/flutter/packages/blob/e6d5f869bf8d5c332e30231424c1493b380c5d4a/packages/pigeon/lib/objc_generator.dart#L265 with this (which is the only other use of `toMap:`). You said "The function can be used in other contexts where it is expected to return a non-null value", but that usage is a function that returns a `(nullable id)`. I don't understand how an explicitly `nullable` function can be expected to return a non-null value; that means you are expecting a function to behave differently than its explicit documented behavior, which is a recipe for disaster. If there is code that is calling `readValueOfType:` but requires that it return a non-null value, then the calling code is wrong and we should fix it.